### PR TITLE
Users getters

### PR DESCRIPTION
**Pull request by nuevoalex**
First cut of the getUTP getter. Still missing raw token balances and some of the additional data in the expected response but getting this in now so the contract changes are available.

### CODE REVIEW

File: packages/augur-state/src/api/Users.ts [hunk 1]
```diff
@@ -1,30 +1,222 @@
...
+  universe?: string,
+  account?: string,
+  marketId?: string,
+  outcome?: number,
+};
+
+export interface GetUserTradingPositionsParams extends UserTradingPositionsParams, SortLimit {
+}
+
+export interface GetProfitLossParams {
```
Comment (adrake33) [inline packages/augur-state/src/api/Users.ts]: This seems like it should extend GetProfitLossSummaryParams to reduce redundancy.

File: packages/augur-state/src/api/Users.ts [hunk 2]
```diff
@@ -1,30 +1,222 @@
...
+  unrealizedCost: string; // denominated in tokens. Cost of shares in netPosition
+  realizedCost: string; // denominated in tokens. Cumulative cost of shares included in realized profit
+  totalCost: string; // denominated in tokens. Always equal to unrealizedCost + realizedCost
+  realizedPercent: string; // realized profit percent (ie. profit/cost)
+  unrealizedPercent: string; // unrealized profit percent (ie. profit/cost)
+  totalPercent: string; // total profit percent (ie. profit/cost)
+  currentValue: string; // current value of netPosition, always equal to unrealized minus frozenFunds
+}
+
+export interface TradingPosition {
```
Comment (adrake33) [inline packages/augur-state/src/api/Users.ts:48]: Might also want to consider having MarketTradingPostion & TradingPosition extend another interface to minimize redundancy.

File: packages/augur-state/src/api/Users.ts [hunk 3]
```diff
@@ -1,30 +1,222 @@
...
+  private readonly db: DB<TBigNumber>;
+
+  constructor(db: DB<TBigNumber>) {
+    this.db = db;
   }
 
-  public async getUserTradingPositions(params: GetUserTradingPositionsParams): Promise<void> {
-    // TODO
+  public async getUserTradingPositions(params: GetUserTradingPositionsParams): Promise<UserTradingPositions> {
+    if (!params.account) {
```
Comment (adrake33) [inline packages/augur-state/src/api/Users.ts]: Weren't we going to handle cases of a missing required value using type definitions? I thought Paul said there was a way to do this using io-ts? 

Regardless, it looks like `account` in the definition for UserTradingPositionsParams should still be changed to not be optional.

File: packages/augur-state/src/api/Users.ts [hunk 4]
```diff
@@ -1,30 +1,222 @@
...
+      throw new Error("'getTradingHistory' requires a 'universe' or 'marketId' param be provided");
+    }
+    const request = {
+      selector: {
+        universe: params.universe,
+        market: params.marketId,
+        account: params.account,
+      },
+      sort: params.sortBy ? [params.sortBy] : undefined,
+      limit: params.limit,
```
Comment (adrake33) [inline packages/augur-state/src/api/Users.ts:94]: `limit` & `offset` are optional, so I think you'll need a check for these as well, similar to the line above.

File: packages/augur-state/src/api/Users.ts [hunk 5]
```diff
@@ -1,30 +1,222 @@
...
+      skip: params.offset,
```
Comment (adrake33) [inline packages/augur-state/src/api/Users.ts:95]: Just a reminder: this function will also need to implement `isSortDescending`.

File: packages/augur-state/src/logs/types.ts [hunk 6]
```diff
@@ -6,10 +6,14 @@ export interface Log {
     transactionIndex: number;
     transactionHash: string;
     logIndex: number;
-    timestamp: number;
```
Comment (adrake33) [inline packages/augur-state/src/logs/types.ts:9]: I realize the log's block timestamp isn't currently getting saved to the DB, but shouldn't it be? For example, `getMarketsInfo` is supposed to return a `creationTimestamp` for the market. Shouldn't we be saving this info to the DB and leaving `timestamp` here?

**Review by adrake33 — APPROVED** [review]
Looks good overall. Just had a few comments/suggestions.

File: packages/augur-state/src/api/Users.ts (hunk 2 above)
Comment (nuevoalex) [reply packages/augur-state/src/api/Users.ts:48]: Yea planning to do this once I get to those getters

File: packages/augur-state/src/api/Users.ts (hunk 3 above)
Comment (nuevoalex) [reply packages/augur-state/src/api/Users.ts]: Ah yea this should be changed to not optional. Np idea on the io-ts though. Something to consider later!

File: packages/augur-state/src/api/Users.ts (hunk 4 above)
Comment (nuevoalex) [reply packages/augur-state/src/api/Users.ts:94]: If they arent included this will just not use them

Comment (nuevoalex) [reply packages/augur-state/src/api/Users.ts:94]: To clarify if the value is undefined they wont apply to the find request

File: packages/augur-state/src/logs/types.ts (hunk 6 above)
Comment (nuevoalex) [reply packages/augur-state/src/logs/types.ts:9]: I'm taking the approach of only including timestamp where needed. We had these in every log in Augur Node by virtue of the architecture so it just got included everywhere. For ordering by time we can just use the id / blocknumber.

Comment (nuevoalex) [reply packages/augur-state/src/logs/types.ts:9]: More specifically unless we have a reason to search or segment by timestamp we shouldn't actually need it.

Comment (adrake33) [reply packages/augur-state/src/logs/types.ts:9]: Ok, looking through the code more, it looks like `creationTimestamp` could probably be removed anyway from getMarketsInfo.

**Pull request merged**